src/EditorFeatures/CSharpTest/Diagnostics/RemoveUnnecessaryCast/RemoveUnnecessaryCastTests.cs

```diff
@@ -3969,5 +3969,42 @@ private static void TakesParams(params object[] foo)
     }
 }");
         }
+
+        [WorkItem(18510, "https://github.com/dotnet/roslyn/issues/18510")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryCast)]
+        public async Task DontRemoveCastOnNegatingEnumValue()
+        {
+            await TestInRegularAndScriptAsync(
```

Comment:
:bulb: I believe the call you want to make is `TestMissingInRegularAndScriptAsync` (meaning no diagnostic is reported).